pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -100,15 +101,26 @@ func (fr *FlowRecords) AddFlowRecordWithoutLock(connKey *flowexporter.Connection
 	fr.recordsMap[*connKey] = *record
 }
 
-// GetFlowRecordFromMap gets the flow record from record map given connection key.
-// This is used only for unit tests.
-func (fr *FlowRecords) GetFlowRecordFromMap(connKey *flowexporter.ConnectionKey) (*flowexporter.FlowRecord, bool) {
+// GetFlowRecordFromMap gets the flow record from record map given the connection key.
+func (fr *FlowRecords) GetFlowRecordFromMap(connKey flowexporter.ConnectionKey) (*flowexporter.FlowRecord, bool) {
```

Comment:
I am not sure that this is totally the same as the pointer version, performance-wise, but it's hard to tell without benchmarking this specific function. Maybe you could run the relevant FlowExporter benchmarks to confirm?

Reply:
Ack the difference between pass by value vs pass by reference. I added a simple benchmark test that tests over 100K records to perform Get and Delete operations. 
Pass by reference run:
`BenchmarkGetRecordAndDeleteFromMap-2   	168049232	         6.737 ns/op	       0 B/op	       0 allocs/op`
Pass by value run:
`BenchmarkGetRecordAndDeleteFromMap-2   	158944249	         7.363 ns/op	       0 B/op	       0 allocs/op`

Because the connection key being a smaller struct, I feel the difference is not very big. Moved to pass by reference as the results show slight benefits.

Reply:
that matches my expectations, thanks for checking